### PR TITLE
[androiddebugbridge] Added mDNS discovery for Fire TV Stick

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added mDNS discovery for Fire TV Stick

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java [hunk 1]
```diff
@@ -361,17 +374,41 @@ public void checkConnection() {
...
+            AndroidDebugBridgeDeviceReadException, ExecutionException {
+        // Add some information about the device
+        try {
+            Map<String, String> editProperties = editProperties();
+            editProperties.put(Thing.PROPERTY_SERIAL_NUMBER, adbConnection.getSerialNo());
+            editProperties.put(Thing.PROPERTY_MODEL_ID, adbConnection.getModel());
+            editProperties.put(Thing.PROPERTY_FIRMWARE_VERSION, adbConnection.getAndroidVersion());
+            editProperties.put(Thing.PROPERTY_VENDOR, adbConnection.getBrand());
+            try {
+                editProperties.put(Thing.PROPERTY_MAC_ADDRESS, adbConnection.getMacAddress());
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java:399]: Unfortunately the `ro.boot.wifimacaddr` property is not available or accessible on all devices but currently it looks like the Mac address is the only unique identifier available in both discovery service implementations.

File: bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/FireTVStickMDNSDiscoveryParticipant.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    @Override
+    public String getServiceType() {
+        return SERVICE_TYPE;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(ServiceInfo service) {
+        if (isAutoDiscoveryEnabled) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/FireTVStickMDNSDiscoveryParticipant.java:85]: Invert condition to reduce nesting?

File: bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/FireTVStickMDNSDiscoveryParticipant.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/FireTVStickMDNSDiscoveryParticipant.java:2]:
```suggestion
 * Copyright (c) 2010-2022 Contributors to the openHAB project
```

File: bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/FireTVStickMDNSDiscoveryParticipant.java (hunk 2 above)
Comment (cweitkamp) [reply bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/FireTVStickMDNSDiscoveryParticipant.java:85]: To be honest I don't see much benefit by doing so.

Comment (jlaur) [reply bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/FireTVStickMDNSDiscoveryParticipant.java:85]: OK.

File: bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidDebugBridgeDiscoveryService.java [hunk 4]
```diff
@@ -61,9 +67,13 @@
     private boolean discoveryRunning = false;
 
     @Activate
-    public AndroidDebugBridgeDiscoveryService(@Reference ConfigurationAdmin admin) {
+    public AndroidDebugBridgeDiscoveryService(final @Reference ConfigurationAdmin admin, //
+            final @Reference TranslationProvider i18nProvider, //
+            final @Reference LocaleProvider localeProvide) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidDebugBridgeDiscoveryService.java:72]: These 2 references are added but not absolutely required?

**Comment by lolodomo** [comment]
Is changing the representation property a breaking change?

File: bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidDebugBridgeDiscoveryService.java (hunk 4 above)
Comment (cweitkamp) [reply bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidDebugBridgeDiscoveryService.java:72]: What is your proposal here. Shall I remove them?

**Comment by cweitkamp** [comment]
> Is changing the representation property a breaking change?

No, I do not think so. It is an internal property which cannot be manipulated the user and if the binding has a proper handling - like I implemented it in this PR - it will be backward compatible.

File: bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidDebugBridgeDiscoveryService.java (hunk 4 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidDebugBridgeDiscoveryService.java:72]: If you agree with me that they are useless, yes I would suggest to not add them.

Comment (cweitkamp) [reply bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidDebugBridgeDiscoveryService.java]: I removed the service references.

**Review by lolodomo — DISMISSED** [review]
LGTM

File: bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java [hunk 5]
```diff
@@ -361,17 +372,41 @@ public void checkConnection() {
...
+            } catch (AndroidDebugBridgeDeviceReadException e) {
+                logger.debug("Refresh properties error: {}", e.getMessage());
+            }
+            ignoreConfigurationUpdate = true;
+            updateProperties(editProperties);
+            ignoreConfigurationUpdate = false;
```
Comment (lolodomo) [inline bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java:403]: @cweitkamp : one last question: why do you need to ignore thing configuration update when you update thing properties? When thing properties are updated, I don't think it is triggering an update of the thing configuration? You are doing something I have never seen in other bindings hamdling properties.

Comment (cweitkamp) [reply bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java:403]: Hm, good question. I guess this is a copy _ paste result from a different binding where the properties are updated during initialization. I can remove this logic here.

Comment (cweitkamp) [reply bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java]: Done.

**Review by lolodomo — APPROVED** [review]
LGTM

**Pull request merged**